analyzers/tests/SonarAnalyzer.TestFramework/Verification/DiagnosticVerifier.cs

```diff
@@ -186,7 +189,9 @@ private static void VerifyBuildErrors(ImmutableArray<Diagnostic> diagnostics, Co
         private static IEnumerable<IssueLocationPair> MatchPairs(CompilationIssues actual, CompilationIssues expected)
         {
             var ret = new List<IssueLocationPair>();
-            foreach (var key in actual.UniqueKeys())
+            // Process project-level issues last to properly match path-based issues first.
+            // Then process primary issues first, so we can update their IssueId for the purpose of matching secondary issues with their IssueId
```

Comment:
I would write this comment in the order of processing:

```csharp
// First, process primary issues to update their IssueId for secondary-matching
// Second, process secondary issues
// Last, process project-level issues that ignore path-based matching

Reply:
I kept it on two lines, and reworded